### PR TITLE
feat(commands): add query user sick days

### PR DESCRIPTION
**Pull request by PatrickDohn**
#### Summary

_Enumerate and describe your changes_

- add command for querying user sick days for the month

#### Checklist

_Double check list and complete these actions before merge_

- [x] PR title is in format `TYPE(scope): description`
- [x] Link PR to Issue (if any issues are related)

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -135,6 +136,17 @@ const onResponse = async (response, isAtMessage = false) => {
         say(`Hello <@${event.user}>! You used ${data.sickUsed} day(s).`);
         break;
       }
+      case GET_THIS_MONTH_SICK: {
+        const data = await GoogleCalendarUtil.getThisMonthsEvents();
+        const sickEvent = `${slackName}: Sick OOO`;
+        const userEvents = _.filter(data, (event) => {
+          if (event.summary.includes(sickEvent)) return event;
```
Comment (davidholyko) [inline src/app.js:143]: this statement works but it doesnt make the most sense.

```
return event.summary.includes(sickEvent)
```

File: src/app.js [hunk 2]
```diff
@@ -135,6 +136,17 @@ const onResponse = async (response, isAtMessage = false) => {
...
+        });
+        console.log(userEvents);
```
Comment (davidholyko) [inline src/app.js:145]: rm

File: src/app.js [hunk 3]
```diff
@@ -135,6 +136,17 @@ const onResponse = async (response, isAtMessage = false) => {
         say(`Hello <@${event.user}>! You used ${data.sickUsed} day(s).`);
         break;
       }
+      case GET_THIS_MONTH_SICK: {
+        const data = await GoogleCalendarUtil.getThisMonthsEvents();
```
Comment (davidholyko) [inline src/app.js:140]: maybe this logic should exist for a user.

```
getThisMonthsEvents(slackName) { 
  // ...
  // returns a list of events for that user
}
```

File: src/commands/commands.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 const TOPICS = {
   GET_SICK: 'GET_SICK',
+  GET_THIS_MONTH_SICK: 'GET_THIS_MONTH_SICK',
```
Comment (davidholyko) [inline src/commands/commands.js:3]: maybe  `GET_THIS_MONTH_SICK_DAYS` makes more sense

File: src/utils/googleCalendarUtil.js [hunk 5]
```diff
@@ -24,6 +24,24 @@ class GoogleCalendarUtil {
     await this.createTodayEvent({ slackName, eventType: VACATION, dateRange });
   }
 
+  async getThisMonthsEvents() {
+    const response = await this.calendar.events.list({
```
Comment (davidholyko) [inline src/utils/googleCalendarUtil.js:28]: we can make a generic `getCalendarEvents` function with params

```
startDate,
endDate
```

File: src/app.js [hunk 6]
```diff
@@ -131,10 +135,11 @@ const onResponse = async (response, isAtMessage = false) => {
       }
       case GET_SICK: {
         // expect slackName to match what the fullName is on the spreadsheet
-        const data = await GoogleSheetUtil.getMyData(slackName);
-        say(`Hello <@${event.user}>! You used ${data.sickUsed} day(s).`);
+        const blocks = SlackBlockKitUtil.buildSearchSelectMenu();
```
Comment (davidholyko) [inline src/app.js:138]: we can remove comment above. out of date

File: src/app.js [hunk 7]
```diff
@@ -257,6 +262,47 @@ app.action(SELECT_PAYMO_TASK_ID, async (response) => {
...
+    const selection = value[key][SELECT_DATE_SEARCH].selected_option.text.text;
+
+    switch (selection) {
+      case MONTH: {
+        const data = await GoogleCalendarUtil.getThisMonthsEvents(slackName);
+        const totalDays = data.length;
+        message = `Hello <@${channel}>! you have taken ${totalDays} sick days this month.`;
+        break;
+      }
+
```
Comment (davidholyko) [inline src/app.js:288]: rm empty line

File: src/app.js [hunk 8]
```diff
@@ -257,6 +262,47 @@ app.action(SELECT_PAYMO_TASK_ID, async (response) => {
...
+      case YEAR: {
+        const data = await GoogleCalendarUtil.getThisYearsEvents(slackName);
+        console.log(data);
+        const totalDays = data.length;
+        message = `Hello <@${channel}>! you have taken ${totalDays} sick days this year.`;
+        break;
+      }
```
Comment (davidholyko) [inline src/app.js:294]: switch statements should have a default

https://www.w3schools.com/js/js_switch.asp

File: src/app.js [hunk 9]
```diff
@@ -257,6 +262,47 @@ app.action(SELECT_PAYMO_TASK_ID, async (response) => {
...
+      case MONTH: {
+        const data = await GoogleCalendarUtil.getThisMonthsEvents(slackName);
+        const totalDays = data.length;
+        message = `Hello <@${channel}>! you have taken ${totalDays} sick days this month.`;
+        break;
+      }
+
+      case YEAR: {
+        const data = await GoogleCalendarUtil.getThisYearsEvents(slackName);
+        console.log(data);
```
Comment (davidholyko) [inline src/app.js]: rm

File: src/app.js [hunk 10]
```diff
@@ -325,12 +369,12 @@ app.action(SUBMIT_DATES, async (response) => {
     const key = _.keys(value);
     const startDate = value[key][SELECT_START_DATE]['selected_date'];
     const endDate = value[key][SELECT_END_DATE]['selected_date'];
-    const validatedDate = GoogleSheetUtil.validateDate(startDate, endDate);
+    const validatedDate = GoogleCalendarUtil.validateDate(startDate, endDate);
```
Comment (davidholyko) [inline src/app.js:371]: 👍

File: src/app.js [hunk 11]
```diff
@@ -325,12 +369,12 @@ app.action(SUBMIT_DATES, async (response) => {
...
     if (validatedDate) {
       const blocks = SlackBlockKitUtil.buildGenericTextBlock('Invalid date');
       chat.postMessage({ channel, blocks });
     } else {
-      await GoogleSheetUtil.createFutureVacationEvent(slackName, startDate, endDate);
+      await GoogleCalendarUtil.createFutureVacationEvent(slackName, startDate, endDate);
```
Comment (davidholyko) [inline src/app.js:376]: 👍

File: src/app.js (hunk 1 above)
Comment (PatrickDohn) [reply src/app.js]: Fixed

File: src/app.js (hunk 3 above)
Comment (PatrickDohn) [reply src/app.js]: Fixed

File: src/commands/commands.js (hunk 4 above)
Comment (PatrickDohn) [reply src/commands/commands.js]: I actually ended not needing that. We just use the get_sick command

File: src/utils/googleCalendarUtil.js (hunk 5 above)
Comment (PatrickDohn) [reply src/utils/googleCalendarUtil.js:28]: Fixed

File: src/app.js (hunk 6 above)
Comment (PatrickDohn) [reply src/app.js:138]: Fixed

File: src/app.js (hunk 7 above)
Comment (PatrickDohn) [reply src/app.js]: Fixed

File: src/app.js (hunk 8 above)
Comment (PatrickDohn) [reply src/app.js:297]: Fixed

File: src/app.js (hunk 9 above)
Comment (PatrickDohn) [reply src/app.js]: Fixed

File: src/app.js [hunk 12]
```diff
@@ -257,6 +262,51 @@ app.action(SELECT_PAYMO_TASK_ID, async (response) => {
...
+      }
+      case YEAR: {
+        const startDate = moment().startOf('year').format();
+        const endDate = moment().endOf('day').format();
+        const data = await GoogleCalendarUtil.getCalendarEvents(slackName, startDate, endDate);
+        const totalDays = data.length;
+        message = `Hello <@${channel}>! you have taken ${totalDays} sick days this year.`;
+        break;
+      }
+      default:
```
Comment (davidholyko) [inline src/app.js:298]: Maybe more descriptve message

```
Selection should  be month or year.
```

File: src/app.js [hunk 13]
```diff
@@ -257,6 +262,51 @@ app.action(SELECT_PAYMO_TASK_ID, async (response) => {
...
+        throw new Error(`Something went really wrong.`);
+    }
+    blocks = SlackBlockKitUtil.buildGenericTextBlock(message);
+    chat.postMessage({ channel, blocks });
+  } catch (error) {
+    console.error(error);
+    const blocks = SlackBlockKitUtil.buildGenericTextBlock([error.toString(), 'Error 409']);
```
Comment (davidholyko) [inline src/app.js:305]: Not sure what 409 refers to... maybe remove

Comment (PatrickDohn) [reply src/app.js]: fixed

File: src/app.js (hunk 12 above)
Comment (PatrickDohn) [reply src/app.js:298]: fixed

File: src/utils/googleCalendarUtil.js [hunk 14]
```diff
@@ -24,6 +24,21 @@ class GoogleCalendarUtil {
...
+      calendarId: GOOGLE_CALENDAR_ID,
+      timeMin: startDate,
+      timeMax: endDate,
+    });
+
+    const data = response.data.items;
+    const sickEvent = `${slackName}: Sick OOO`;
+    const userEvents = _.filter(data, (event) => {
+      if (event.summary.includes(sickEvent)) return true;
```
Comment (davidholyko) [inline src/utils/googleCalendarUtil.js:37]:
```
return event.summary.includes(sickEvent)
```

**Review by davidholyko — APPROVED** [review]

**Pull request merged**